### ISSUE TITLE
Ionic serve error: Error: ENOENT: no such file or directory, open '/home/.../IonicTest/www/index.html'

### ISSUE THREAD

**Issue by pdrosos** [driftyco/ionic-cli#1420]
Hi,

I have just removed the previous version of ionic cli and installed the new one:
npm uninstall -g ionic
npm cache clean
npm install -g ionic

and started a new project:
ionic start IonicTest --v2

The project is generated with the latest ionic-angular 2.0.0-rc.0, but when I go to the project directory and run ionic serve, I see an error in the browser:
Error: ENOENT: no such file or directory, open '/home/pavlina/IonicTest/www/index.html'

CC @jgw96 @danbucholtz @brandyscarney

**Comment by flutterwhale** [comment]
+1

same problem

**Comment by joshuamorony** [comment]
Try updating to the current version of Node, not LTS: https://nodejs.org/en/download/current/

I had trouble on 4.x, but with Node v6.7.0 and npm v3.10.3 everything worked fine.

**Comment by pdrosos** [comment]
@joshuamorony thanks! It works with Node 6.7.0 indeed. I was with 6.5.0 before, which is not the LTS, but there was obviously some issues with it.

What about the gulpfile.json? It's still missing. What is the way to configure the build tasks now? Or probably webpack.config?

**Comment by huichogil** [comment]
same problem 
node5.12.0
npm 3.8.6

**Comment by chandandas** [comment]
I have done everything mentioned in this forum but still has same issue.

This is very frustrating.

**Comment by adirzoari** [comment]
save problem

**Comment by mmmathur** [comment]
I've node 6.6.0, npm 3.10.3 and ionic 2.0.0. I'm getting same error too. I also noticed that there is no gulpfile.js to build the app.
Please help

**Comment by adirzoari** [comment]
I just wake up this morning with many problems with ionic 2 project. yesterday everything worked project with another project I worked on.
anyone know how to import "LocalStorage"? it shows me error
"LocalStorage is undefined".

**Comment by brandyscarney** [comment]
A few things need to be checked:

1. CLI version should be 2.1.0 **not beta**. If you run `ionic -v` and have anything other than `2.1.0`, run:

    ```
    npm uninstall -g ionic
    npm install -g ionic
    ```

2. NPM version needs to be 3.x, you can check this by running `npm -v`, if it is not, upgrade your npm version by installing Node 6 or following the instructions here: https://docs.npmjs.com/getting-started/installing-node#updating-npm

We released RC0 with many breaking changes. These have been outlined in the [changelog.](https://github.com/driftyco/ionic/blob/master/CHANGELOG.md#200-rc0-2016-09-28) Please check the changelog and let me know if we need to add clarification to it. We also posted a blog here: http://blog.ionic.io/announcing-the-ionic-2-release-candidate/

@adirzoari Storage has been moved. See the [storage](https://github.com/driftyco/ionic/blob/master/CHANGELOG.md#storage) section in the changelog.

**Comment by adirzoari** [comment]
@brandyscarney thank your for you reply.
what about the LocalStorage? how I import it?
I just trying to do 
`  public local: Storage;
    this.local=new Storage(LocalStorage);

;`
and I get error--> LocalStorage is undefined

**Comment by adirzoari** [comment]
@brandyscarney  another thing.
when I try to install npm install @ionic/storage --save --save-exact
I get this error

> npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.14

**Comment by danbucholtz** [comment]
@adirzoari,

LocalStorage was removed from Ionic as it wasn't a best practice.  See [the changelog](https://github.com/driftyco/ionic/blob/master/CHANGELOG.md#storage) for instructions on how to get up and running with the new `@ionic/storage` module.

Thanks,
Dan

**Comment by blackkaiken** [comment]
i cant solve the Eonent issue for 4 days i am about to turn mad :(

**Comment by danbucholtz** [comment]
@blackkaiken,

Can you provide some more info?

If you run `ionic --version`, what prints out?  If you run `npm --version`, what prints out?

Can you run `npm run build` from your project directory and let me know the output?

Thanks,
Dan

**Comment by adirzoari** [comment]
hello, why I don't have access to global.scss?
I mean when I defined button with primary color it's not change the color to primary color, why is that?
i checked the global page exist.

**Comment by danbucholtz** [comment]
@adirzoari,

Please create separate issues for your issues.

https://github.com/driftyco/ionic/issues

Thanks,
Dan

**Comment by adirzoari** [comment]
hey dan, I opened a new issue. just saying that I never get recive to my 4 posts i open.. just when I write into other issues I get recive.. feel confused about that.

**Comment by blackkaiken** [comment]
@danbucholtz @adirzoari 

thank you very much for your responses here is the screen shot. if you help me i would be honoured

http://imgim.com/image/4108inciy4859929.png

**Comment by lucascardoso** [comment]
I have the same problem
![error_app_test3](https://cloud.githubusercontent.com/assets/2519428/19151530/845115e4-8ba3-11e6-89aa-5312ef32d484.png)

ionic info:
![error_app_test3](https://cloud.githubusercontent.com/assets/2519428/19151548/9f4593b6-8ba3-11e6-9832-71e0172d3fec.png)

**Comment by danbucholtz** [comment]
Are you all running `npm 3.x`?

Can you run `npm --version` and verify?

Thanks,
Dan

**Comment by lucascardoso** [comment]
Yes!
![captura de tela de 2016-10-06 11-40-11](https://cloud.githubusercontent.com/assets/2519428/19156868/b29fb1f6-8bb9-11e6-8406-f836c02f0169.png)

Thank you!

**Comment by danbucholtz** [comment]
@lucascardoso, Thanks!

Can you run `npm run build` and give me the entire console output? I don't have enough to work off of right now.

Thanks,
Dan

**Comment by lucascardoso** [comment]
@danbucholtz, this is the output.
![captura de tela de 2016-10-06 13-08-22](https://cloud.githubusercontent.com/assets/2519428/19160421/323c17fe-8bc6-11e6-8640-287253b904e5.png)

**Comment by danbucholtz** [comment]
@lucascardoso,

What do your `package.json` `scripts` section look like?

Thanks,
Dan

**Comment by danbucholtz** [comment]
It looks like you may have installed `npm` with `ignore-scripts`.

http://blog.npmjs.org/post/141702881055/package-install-scripts-vulnerability

Thanks,
Dan

**Comment by lucascardoso** [comment]
@danbucholtz, 
apparently this normal for package.json file.
I have this script installed ignore.
![captura de tela de 2016-10-06 13-29-12](https://cloud.githubusercontent.com/assets/2519428/19161243/67bc7632-8bc9-11e6-96e8-b9dccb92da1d.png)

Thank you!

**Comment by lucascardoso** [comment]
@danbucholtz, 
It should create the index.html file when running the command ``` ionic serve``` the first time?

This is the log that appears when running the ``` ionic serve```
![captura de tela de 2016-10-06 13-51-46](https://cloud.githubusercontent.com/assets/2519428/19162077/c0535646-8bcc-11e6-8858-235b704aff95.png)

Thank you!

**Comment by danbucholtz** [comment]
Your issues are likely do to your setup. Can you run `npm run build`? That will build everything and get it to a good state. If it doesn't run, the script ignore thing is still not quite right.

Thanks,
Dan

**Comment by lucascardoso** [comment]
@danbucholtz,
thank you so much! It worked for me! :smile: 
I ran the command ``` npm config set ignore-scripts false```
and then I ran the command ```npm run build```
and worked as you had shown before
```
> ionic-hello-world@ build /home/lucas-cardoso/code/ionic/test5
> ionic-app-scripts build

[14:29:05]  ionic-app-scripts 0.0.30
[14:29:05]  build prod started ...
[14:29:05]  clean started ...
[14:29:05]  clean finished in 10 ms
[14:29:05]  copy started ...
[14:29:05]  ngc started ...
[14:29:05]  lint started ...
[14:29:05]  copy finished in 50 ms
[14:29:05]  lint finished in 633 ms
[14:29:20]  ngc finished in 15.59 s
[14:29:20]  bundle prod started ...
[14:29:27]  bundle prod finished in 7.33 s
[14:29:27]  sass started ...
[14:29:29]  sass finished in 1.33 s
[14:29:29]  minify started ...
[14:29:29]  cleancss started ...
[14:29:29]  uglifyjs started ...
[14:29:42]  uglifyjs finished in 12.82 s
[14:29:42]  cleancss finished in 12.82 s
[14:29:42]  minify finished in 12.82 s
[14:29:42]  build prod finished in 37.09 s
```

**Comment by blackkaiken** [comment]
@danbucholtz sir i cannot even start app these didnt work for me thank you.`
![gtttt](https://cloud.githubusercontent.com/assets/20086535/19170755/4b044fac-8c22-11e6-9d9a-90e5adbe14d4.png)
`

**Comment by blackkaiken** [comment]
sir i cannot even start app these didnt work for me thank you
![gtttt](https://cloud.githubusercontent.com/assets/20086535/19170859/b052555c-8c22-11e6-913d-a8ef0a967b31.png)

**Comment by blackkaiken** [comment]
@danbucholtz sir these didnt for me  icant even start app. could you help me. i cant start ionic 2 for 8 days. this must be joke..
![gtttt](https://cloud.githubusercontent.com/assets/20086535/19170930/f9631614-8c22-11e6-8e27-8044fbe91bf9.png)

**Comment by blackkaiken** [comment]
@danbucholtz  sir these didnt work for me.. could you help me please?

![gittt](https://cloud.githubusercontent.com/assets/20086535/19171025/76b75616-8c23-11e6-820a-ec6eb8e1e780.png)

**Comment by danbucholtz** [comment]
@jthoms1, ^, what do you think of that?

CC: @blackkaiken 

Thanks,
Dan

**Comment by blackkaiken** [comment]
@jthoms1
![adsiz11](https://cloud.githubusercontent.com/assets/20086535/19207000/a2e1bbb4-8cf5-11e6-8943-96b9745ee258.png)

here is ionic info also... Does anyone help me? 

one recycler view on native android with tons of code ... i want ionic with angular 2 :)

**Comment by blackkaiken** [comment]
"npm ls -g ionic" here is the  results.. 
![ionic11](https://cloud.githubusercontent.com/assets/20086535/19212091/d230a3f0-8d52-11e6-9ba9-5638dc8cf16a.png)

**Comment by kkganesan** [comment]
`
> ionic-app-scripts watch

[22:12:26]  ionic-app-scripts 0.0.33
[22:12:26]  watch started ...
[22:12:26]  build dev started ...
[22:12:26]  clean started ...
[22:12:26]  clean finished in 11 ms
[22:12:26]  copy started ...
[22:12:26]  transpile started ...
[22:12:26]  lint started ...
[22:12:28]  lint finished in 1.75 s
[22:12:30]  transpile finished in 3.88 s
[22:12:30]  bundle started ...
[22:12:30]  copy finished in 3.96 s
[22:12:40]  bundle finished in 10.36 s
[22:12:40]  sass started ...
[22:12:42]  sass finished in 2.04 s
[22:12:42]  build dev finished in 16.31 s
[22:12:42]  watch ready in 16.39 s
Running live reload server: http://localhost:35729
Watching: www/**/*, !www/lib/**/*, !www/**/*.map
√ Running dev server:  http://localhost:8100
Ionic server commands, enter:
  restart or r to restart the client app from the root
  goto or g and a url to have the app navigate to the given url
  consolelogs or c to enable/disable console log output
  serverlogs or s to enable/disable server log output
  quit or q to shutdown the server and exit
`

ionic serve -->
In browser Getting Error: ENOENT: no such file or directory, open ..index.html

www contains 
assets			build			index.html		manifest.json		service-worker.js

ionic -v 2.1.1
node 6.7.0
npm 3.10.3

**Comment by speedfl** [comment]
Solution!!!!!!!!!!!!!!!!!!!!!!!!

"scripts": {
    "build": "ionic-app-scripts build",
    "watch": "ionic-app-scripts watch",
    "serve:before": "watch",
    "emulate:before": "build",
    "deploy:before": "build",
    "build:before": "build",
    "run:before": "build"
  }

in your package.json

**Comment by gopi2mohan** [comment]
@speedfl thanks it solved for me. but this is not getting added by default when we create project.

**Comment by sklaujb** [comment]
Hi, I am still having this problem and is really frustrating.......any can help ? 
This is my ionic info

Cordova CLI: 6.3.1
Ionic Framework Version: 2.0.0-rc.3
Ionic CLI Version: 2.1.0
Ionic App Lib Version: 2.1.0-beta.1
ios-deploy version: Not installed
ios-sim version: 5.0.6 
OS: Mac OS X El Capitan
Node Version: v7.0.0
Xcode version: Xcode 7.3.1 Build version 7D1014

**Comment by speedfl** [comment]
@sklaubjb can you show your package.json

**Comment by sklaujb** [comment]
Hello speedfl, this is the package.json

{
  "name": "ionic-hello-world",
  "author": "Ionic Framework",
  "homepage": "http://ionicframework.com/",
  "private": true,
  "scripts": {
   "ionic:build": "ionic-app-scripts build",
    "ionic:serve": "ionic-app-scripts serve"
  },
  "dependencies": {
    "@angular/common": "2.1.1",
    "@angular/compiler": "2.1.1",
    "@angular/compiler-cli": "2.1.1",
    "@angular/core": "2.1.1",
    "@angular/forms": "2.1.1",
    "@angular/http": "2.1.1",
    "@angular/platform-browser": "2.1.1",
    "@angular/platform-browser-dynamic": "2.1.1",
    "@angular/platform-server": "2.1.1",
    "@ionic/storage": "1.1.6",
    "ionic-angular": "2.0.0-rc.3",
    "ionic-native": "2.2.3",
    "ionicons": "3.0.0",
    "rxjs": "5.0.0-beta.12",
    "zone.js": "0.6.26"
  },
  "devDependencies": {
    "@ionic/app-scripts": "0.0.45",
    "typescript": "2.0.6"
  },
  "cordovaPlugins": [
    "cordova-plugin-device",
    "cordova-plugin-console",
    "cordova-plugin-whitelist",
    "cordova-plugin-splashscreen",
    "cordova-plugin-statusbar",
    "ionic-plugin-keyboard"
  ],
  "cordovaPlatforms": [
    "ios",
    {
      "platform": "ios",
      "version": "",
      "locator": "ios"
    }
  ],
  "description": "NewdotScanner: An Ionic project"
}

**Comment by gopi2mohan** [comment]
@speedfl i too getting the same issue. my package.json is also same as @sklaujb  but if i update my "scripts": {
"ionic:build": "ionic-app-scripts build",
"ionic:serve": "ionic-app-scripts serve"
},

into this

 "scripts": {
"build": "ionic-app-scripts build",
"watch": "ionic-app-scripts watch",
"serve:before": "watch",
"emulate:before": "build",
"deploy:before": "build",
"build:before": "build",
"run:before": "build"
} 

everything works fine but why it is not automatically generated by ionic when creating the project

**Comment by speedfl** [comment]
to be honnest I don't know I am not part of ionic team. I am just a developer like you.

**Comment by gopi2mohan** [comment]
@speedfl  no problem. but after doing some research i found this blog http://blog.ionic.io/improvements-to-ionic-build-process/ everything looks ok but not able to see the gulp file run before lite-server runs.

**Comment by danbucholtz** [comment]
It seems like the build is failing and note reporting the outcome correctly.

Can you update the following packages so we can troubleshoot this better?

```
npm install -g ionic@latest
npm install @ionic/app-scripts@latest
```

Can you also append a `--debug` to the npm script for `ionic:serve` so we can see additional logging? That may provide a hint.

Thanks,
Da

**Comment by gopi2mohan** [comment]
@danbucholtz I got the out put i am using ubuntu16.04, So i forgot to use sudo command.


Now it is working fine thanks

**Comment by sklaujb** [comment]
I am doing everything I can get from here but is still facing the same issue. I am also having build error when trying to build for android. This is what I get

_

> cache taskArtifacts.bin (/Volumes/My Passport/ionic/fucku/platforms/android/.gradle/2.13/taskArtifacts/taskArtifacts.bin) is corrupt. Discarding.
> _

_

> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Could not open cache taskArtifacts.bin (/Volumes/My Passport/ionic/fucku/platforms/android/.gradle/2.13/taskArtifacts/taskArtifacts.bin).
> > Corrupted FreeListBlock 0 found in cache '/Volumes/My Passport/ionic/fucku/platforms/android/.gradle/2.13/taskArtifacts/taskArtifacts.bin'.
> 
> * Try:
> 
> Run with --stacktrace option to get the stack trace. Run with 
> --info or --debug option to get more log output.
> 
> BUILD FAILED

_

Are this 2 issues related? I am really at a lost of what to do next.............any advice pls.........

**Comment by sklaujb** [comment]
No one ??...........

**Comment by kevendra** [comment]
please try
$ npm install ionic@latest -g
$ npm install @ionic/app-scripts@latest --save-dev
$ npm run ionic:build
$ npm run ionic:serve

**Comment by gigetto03** [comment]
i also have the same problem, i have tried to send the above commands, i attach the screen of error received (screen is referred to "npm install @ionic/app-scripts@latest")
![screenshot 10](https://cloud.githubusercontent.com/assets/24192882/20642058/0219d544-b406-11e6-878b-27759d40ca5f.png)

**Comment by paul-landaeta-tierconnect-com** [comment]
I solved with npm run ionic:build

**Comment by ShiroTeky** [comment]
Thanks @danbucholtz   After one week of frustration. You save my day. Merry Xmas

**Comment by Elrashid** [comment]
solved by running 
npm run build
Before
ionic serve

**Comment by amity203** [comment]
I am getting this error, despite of trying all possible steps shown here, writing here after frustrated 2 hours:
Runtime Error
Module build failed: Error: ENOENT: no such file or directory, open '/Users/Documents/IonicMaster/src/pages/menuthree/menuthree.js' at Error (native)
Stack
Error: Module build failed: Error: ENOENT: no such file or directory, open '/Users/Documents/IonicMaster/src/pages/menuthree/menuthree.js'
    at Error (native)
    at Object.<anonymous> (http://localhost:8100/build/main.js:97312:7)
    at __webpack_require__ (http://localhost:8100/build/main.js:20:30)
    at Object.<anonymous> (http://localhost:8100/build/main.js:56476:79)
    at __webpack_require__ (http://localhost:8100/build/main.js:20:30)
    at Object.<anonymous> (http://localhost:8100/build/main.js:97232:82)
    at __webpack_require__ (http://localhost:8100/build/main.js:20:30)
    at Object.<anonymous> (http://localhost:8100/build/main.js:85905:73)
    at __webpack_require__ (http://localhost:8100/build/main.js:20:30)
    at Object.<anonymous> (http://localhost:8100/build/main.js:105203:70)
    at __webpack_require__ (http://localhost:8100/build/main.js:20:30)
Ionic Framework: 2.0.0-rc.5
Ionic Native: 2.2.11
Ionic App Scripts: 1.0.0
Angular Core: 2.2.1
Angular Compiler CLI: 2.2.1
Node: 6.9.4
OS Platform: macOS Sierra
Navigator Platform: MacIntel
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36

Anyone know of any other solution, other then what is mentioned here.

**Comment by amity203** [comment]
Why it is looking for a .JS file instead of a .TS file

**Comment by Endtry** [comment]
$ npm install ionic@latest -g
$ npm install @ionic/app-scripts@latest --save-dev

the two commands above also fixes the live reload issue.

**Comment by chpecson** [comment]
I do run the build first and then execute ionice serve. And it works fine for me.
`sh`
npm run build
ionic ser

**Comment by gacalves** [comment]
`npm run build` in project folder solved for me.

`npm -v
3.10.10
$ cordova -v
6.5.0
$ ionic -v
2.2.1
$ node -v
v6.10.0`

**Comment by MorisNotari** [comment]
In my case the problem was caused by the fact that my project was in a subfolder of a **NTSF junction** on Windows. 

E.g.: `c:/works/my.ionic.project`

Where `works` is a NTFS junction to a directory on the same drive.

`npm run build` cannot find all necessary files but doesn't report any error to the user.

Don't know if this is a NPM issue or a Ionic CLI issue, maybe @danbucholtz (or others Ionic team members) could figure it out...

**Comment by yhalwani** [comment]
I have had similar errors 
"Runtime Error
Module build failed: Error: ENOENT: no such file or directory, open 'C:\Users\Youssef\Documents\App Dev\Ionic\Ionic CLI\Samples\tab-sample\tabApp3\src\app\main.js' at Error (native)"

My ionic info
```
Cordova CLI: 6.5.0
Ionic Framework Version: 3.0.1
Ionic CLI Version: 2.2.2
Ionic App Lib Version: 2.2.1
Ionic App Scripts Version: 1.3.3
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Windows 8.1
Node Version: v6.10.2
Xcode version: Not installed
```

I have tried running 
```
npm install -g ionic@latest
npm install @ionic/app-scripts@latest
npm run ionic:build
npm run ionic:serve --debug
```

Here is the debug output that I think is causing the issue 

![debug_output](https://cloud.githubusercontent.com/assets/6643501/25093120/c4290cdc-235e-11e7-802c-be8a78e41759.PNG)

Basically the wepack loader is failing to load main.js
Any help is appreciated!


@jgw96 @danbucholtz @brandyscarney

**Comment by sjosen** [comment]
@amity203 I had the same issue until I realised I was importing a component with a small typo

    import {User} from "../models/user"

instead of

    import {User} from "../models/User"

It was first when running ionic run android --device --prod I was presented with an error. The app worked without issues in the serve environment

**Comment by SanjayYadav90** [comment]
Actually we are missing index.html file here-
`www/index.html`
just copy and paste index file to resolve this issue-
copy form `src/index.html`
paste here `www/index.html`

hopefully that will for for you as this worked for me.

**Comment by prasad456** [comment]
Hi having while creating a project

my installed versions are
node 8.6.0
nodejs 8.6.0
npm 5.3.0

what should I do?

**Comment by Nigrito** [comment]
Hi, I am still having this problem and is really VERY frustrating.......any can help ?

![capture](https://user-images.githubusercontent.com/31864843/31842715-5ee85dd4-b5e7-11e7-9705-889cd4e9687a.PNG)

my ionic info 
`
cli packages: (C:\Users\monako\AppData\Roaming\npm\node_modules)

    @ionic/cli-utils  : 1.13.1
    ionic (Ionic CLI) : 3.13.2

global packages:

    cordova (Cordova CLI) : 7.1.0

local packages:

    @ionic/app-scripts : 3.0.0
    Cordova Platforms  : android 6.3.0
    Ionic Framework    : ionic-angular 3.7.1

System:

    Android SDK Tools : 25.2.5
    Node              : v8.7.0
    npm               : 2.15.12
    OS                : Windows 10

Misc:

    backend : pro`

**Comment by seanbrice** [comment]
If it helps anyone, in my case the error similar to that mentioned was caused by having the following in a **block commented** ( as in inside..  /*  */  ) section of code. The comment was in a Provider, and had nothing to do with the boiler plate setup.. it was just being parked towards the base of my script as reference for something I was working on.

	@Component({

	selector: 'page-home',
	templateUrl: 'home.html'
	})

I narrowed it down to removing this chunk of code causing the issue to disappear, even though there was a still a lot of other code left in the block comment.

There was no use of any build command between removing and putting it back, seeing the error disappear, then appear again when the code was present.. just running **ionic serve --platform="ios"**

So, worthwhile checking to see if you have anything lingering around in your comments that is magically being plucked out and processed at build time. If there's a reason why the above is meant to happen, i.e. the block comment is seemingly ignored, I'd be interested to know.
